### PR TITLE
fix(ui-react): ariaLabel now optional on Icon component, update Icon docs

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes

This PR makes the `ariaLabel` prop optional on the Icon component.

**Context**
A common use case for an Icon is using it decoratively. It feels a little awkward passing an empty `aria-label` to Icon when you still need to add `aria-hidden="true"` to make it hidden in the accessibility tree. Essentially:
```
<Icon ariaLabel=""  aria-hidden="true"   ... />
```
Additionally, your custom SVG could have a title element, which means you would previously need to duplicate that title text as the ariaLabel or risk losing the label.
```
<Icon ariaLabel="Search">
  <title>Search</title>
</Icon>
```

Changing the prop to be optional provides more flexibility.

**Additionally, this PR** 
- updates the tests to add an ariaLabel specific test, and remove the ariaLabel from the other tests.
- updates the Icon demo to match the format the other components use. 
- adds (hopefully?) a better way to show how width/height/viewBox affect the demo.
- updates the Accessibility section to show a few considerations for accessible icons. 
- fixes an Alert that had code elements in it that were causing a scroll on mobile.

**Old Demo**

**New Demo**

**Broken Alert**
Screen Shot 2022-07-27 at 10 17 45 PM

**Fixed Alert**
Screen Shot 2022-07-27 at 10 19 33 PM

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/components/icon/demo.tsx [hunk 1]
```diff
@@ -1,96 +1,64 @@
...
-        onChange={(event) => setAriaLabel(event.target.value)}
-      />
-      <TextField
-        label="color"
-        value={color}
-        onChange={(event) => setColor(event.target.value)}
-      />
-    </View>
+    `<Icon` +
+    (pathData ? `\n  pathData=${JSON.stringify(pathData)}` : '') +
```
Comment (joebuono) [inline docs/src/pages/[platform]/components/icon/demo.tsx:19]: Just FYI, we have a `getPropString` utility function for handling this in `docs/src/pages/[platform]/components/utils/getPropString.ts`

File: docs/src/pages/[platform]/components/icon/react.mdx [hunk 2]
```diff
@@ -175,7 +180,10 @@ Use the `color` prop to change the Icon color. The fill of the path inside the S
 
 ## Accessibility
 
-Set an `aria-label` attribute for an Icon.
+The Icon component does not require a label by default because there are a number of ways to use an Icon in an accessible way:
+- You can set an `aria-label` attribute on the Icon
+- You can use a `<title></title>` element when passing SVG elements as the child of the Icon
+- You can use the Icon decoratively, when a label would be redundant.
```
Comment (joebuono) [inline docs/src/pages/[platform]/components/icon/react.mdx:186]: This a11y section is excellent! I didn't know about the `<title>` use case

**Review by joebuono — APPROVED** [review]
LGTM! Thank you for also updating the Icon demo

File: docs/src/pages/[platform]/components/icon/demo.tsx (hunk 1 above)
Comment (hbuchel) [reply docs/src/pages/[platform]/components/icon/demo.tsx]: Nice! Updated to use that where I could in latest.

**Review by joebuono — APPROVED** [review]

**Review by jacoblogan — APPROVED** [review]

**Pull request merged**